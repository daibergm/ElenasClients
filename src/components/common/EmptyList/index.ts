export { default as EmptyList } from './EmptyListComponent';
