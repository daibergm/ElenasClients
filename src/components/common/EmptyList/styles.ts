import { StyleSheet } from 'react-native';

import { Colors } from '../../../assets';

export default StyleSheet.create({
  container: {
    alignItems: 'center',
  },
  title: {
    marginTop: 20,
    fontSize: 15,
    fontWeight: '600',
  },
  description: {
    marginTop: 12,
    fontSize: 15,
    color: Colors.grey,
  },
});
