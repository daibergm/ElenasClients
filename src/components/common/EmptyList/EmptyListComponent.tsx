import React from 'react';
import { View, Text } from 'react-native';

// @Constants
import { TEST_IDS } from '../../../constants';

import styles from './styles';

interface Props {
  description: string;
}

function EmptyListComponent({ description }: Props) {
  return (
    <View testID={TEST_IDS.emptyList} style={styles.container}>
      <Text style={styles.title}>Buscar Clientes</Text>
      <Text style={styles.description}>{description}</Text>
    </View>
  );
}

export default EmptyListComponent;
