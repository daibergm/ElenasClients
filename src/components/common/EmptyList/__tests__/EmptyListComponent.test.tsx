import React from 'react';
import { render } from '@testing-library/react-native';

// @Components
import EmptyList from '../EmptyListComponent';

describe('EmptyList test suits', () => {
  it('Render correctly', () => {
    const emptyList = render(<EmptyList description="" />);
    expect(emptyList).toMatchSnapshot();
  });
});
