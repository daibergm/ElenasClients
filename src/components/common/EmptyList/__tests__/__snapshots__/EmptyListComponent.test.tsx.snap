// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyList test suits Render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
  testID="emptyList"
>
  <Text
    style={
      Object {
        "fontSize": 15,
        "fontWeight": "600",
        "marginTop": 20,
      }
    }
  >
    Buscar Clientes
  </Text>
  <Text
    style={
      Object {
        "color": "grey",
        "fontSize": 15,
        "marginTop": 12,
      }
    }
  >
    
  </Text>
</View>
`;
